src/app/server/Server.cpp

```diff
@@ -161,6 +162,12 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     SetAttributePersistenceProvider(&mAttributePersister);
     SetSafeAttributePersistenceProvider(&mAttributePersister);
 
+    // Ember requires a persistence provider:
```

Comment:
I don't quite understand this comment.  It's talking about "requires a persistence provider" as if that's the thing it's documenting, but it's documenting the SetDataModelProvider() call....

It seems like the actual ordering constraints here are:

1. InitDataModelHandler requires a data model provider (why?  Where does it use it?  This part is unclear to me)
2. CodegenDataModelProvider requires a persistence provider (but it does not use it directly.... the actual use likely relevant here is inside code called directly from InitDataModelHandler, not via the provider.  The provider _can_ call things that need persistence, but I don't think it does that as part of the startup sequence.)

Reply:
Updated the comment. We essentially want to move all ember specific init in the data model provider since that is our "interface to ember". However until then, the ordering in Server.cpp is very intentional.

The call stack is generally "load defaults" which may load persisted NVM values it seems:

```
InitDataModelHandler -> emberAfInit -> emberAfInitializeAttributes -> emAfLoadAttributeDefaults
```